src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageModelActivatorProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
```

Comment:
If we have a `.Infrastructure` namespace already parked it would be good to put this there and document it for extensibility. It was common in the past to subclass our factories.